dns/dns_updater.go

```diff
@@ -167,18 +185,39 @@ func (u *updater) Update(entries controller.IngressEntries) error {
 	return nil
 }
 
-func (u *updater) determineManagedRecordSets(rrs []*route53.ResourceRecordSet) []*route53.ResourceRecordSet {
+func (u *updater) flattenRoute53ResourceRecordSet(rrs []*route53.ResourceRecordSet) []record {
+	var records []record
+
+	for _, recordSet := range rrs {
+		if len(recordSet.ResourceRecords) == 1 {
+			records = append(records, record{
+				name:     *recordSet.Name,
+				pointsTo: *recordSet.ResourceRecords[0].Value,
+			})
+		} else {
+			records = append(records, record{
+				name:            *recordSet.Name,
+				pointsTo:        *recordSet.AliasTarget.DNSName,
+				aliasHostedZone: *recordSet.AliasTarget.HostedZoneId,
+			})
+		}
+	}
+
+	return records
+}
+
+func (u *updater) determineManagedRecordSets(rrs []record) []record {
 	managedLBs := make(map[string]bool)
 	for _, dns := range u.schemeToDNS {
 		managedLBs[dns.dnsName] = true
 	}
-	managed := make([]*route53.ResourceRecordSet, 0)
-	nonManaged := make([]string, 0)
+	var managed []record
+	var nonManaged []string
 	for _, rec := range rrs {
-		if rec.AliasTarget != nil && rec.AliasTarget.DNSName != nil && managedLBs[*rec.AliasTarget.DNSName] {
+		if rec.name != "" && managedLBs[rec.pointsTo] {
```

Comment:
This logic seems wrong to me - won't it will break if anyone has non alias records in their zone? We need to support any arbitrary record. So this would need to check if it matches one of our static hostnames.

Reply:
Arbitrary record types wasn't within the scope but I think where we flatten/convert the records could be made much clearer to show what records we are supporting (as referenced by https://github.com/sky-uk/feed/pull/153/files/11f4334038e55b10b9358fab4dd81f44a90ee8e7#r135288514)